### PR TITLE
[drone] Add RBAC support

### LINKED ISSUE #4143: [incubator/drone] RBAC support
**Is this a request for help?**: Yes

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature request

**Version of Helm and Kubernetes**: 

```
$ helm version
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-09T21:51:54Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.4+coreos.0", GitCommit:"4292f9682595afddbb4f8b1483673449c74f9619", GitTreeState:"clean", BuildDate:"2017-11-21T17:22:25Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**: [`incubator/drone`](https://github.com/kubernetes/charts/tree/master/incubator/drone)

**What happened**: I received an error while attempting to deploy a container to my cluster from within Drone.

```
Error from server (Forbidden): deployments.extensions "api" is forbidden: User "system:serviceaccount:bananas:default" cannot get deployments.extensions in the namespace "bananas"
```

**What you expected to happen**: I expected the Drone chart to automatically setup RBAC when prompted with an `rbac` option in `values.yaml`, similar to [`stable/kube-lego`](https://github.com/kubernetes/charts/tree/master/stable/kube-lego).

**How to reproduce it** (as minimally and precisely as possible): Create a cluster with RBAC and attempt to call the Kubernetes API.

**Anything else we need to know**: I am willing to start working on this feature if nobody else already has 👍

### PR DESCRIPTION
**Pull request by puradox**
**What this PR does / why we need it**:
RBAC support is vital for using Drone for continuous delivery on a Kubernetes stack that is running in RBAC mode. [drone-kubernetes][2] is one such plugin that will not work on RBAC-enabled Kubernetes stack with the current version of this chart.

**Which issue this PR fixes**: fixes #4143 

**Special notes for your reviewer**:
These changes add RBAC support to Drone by creating a Service Account, Role, and Role Binding. Many of these templates were taken from the more popular [nginx-ingress][3] chart, which implements RBAC support well.

[2]: https://github.com/honestbee/drone-kubernetes
[3]: https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress

### CODE REVIEW

File: incubator/drone/values.yaml [hunk 1]
```diff
@@ -178,3 +178,12 @@ persistence:
 ## the agents and servers, otherwise this will be auto-generated.
 ##
 # sharedSecret: supersecret
+
+## Install Default RBAC roles and bindings
+rbac:
```
Comment (timstoop) [inline incubator/drone/values.yaml]: Can you add these to the README as well, please?

**Review by timstoop — CHANGES_REQUESTED** [review]

File: incubator/drone/values.yaml (hunk 1 above)
Comment (puradox) [reply incubator/drone/values.yaml]: Nice catch. I'll add those entries in right now

**Comment by unguiculus** [comment]
Please follow our RBAC best practices:
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

/hold
/assign

File: incubator/drone/templates/role-binding.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{{ if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/{{ required "A valid .Values.rbac.apiVersion entry required!" .Values.rbac.apiVersion }}
+kind: RoleBinding
+metadata:
+  name: {{ template "drone.fullname" . }}
+  labels:
+    app: {{ template "drone.fullname" . }}
```
Comment (unguiculus) [inline incubator/drone/templates/role-binding.yaml]: `app: {{ template "drone.name" . }}`

File: incubator/drone/templates/role.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{{ if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/{{ required "A valid .Values.rbac.apiVersion entry required!" .Values.rbac.apiVersion }}
+kind: Role
+metadata:
+  name: {{ template "drone.fullname" . }}
+  labels:
+    app: {{ template "drone.fullname" . }}
```
Comment (unguiculus) [inline incubator/drone/templates/role.yaml]: `app: {{ template "drone.name" . }}`

File: incubator/drone/templates/service-account.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{{ if .Values.serviceAccount.create }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "drone.serviceAccountName" . }}
+  labels:
+    app: {{ template "drone.fullname" . }}
```
Comment (unguiculus) [inline incubator/drone/templates/service-account.yaml]: `app: {{ template "drone.name" . }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/drone/templates/service-account.yaml (hunk 4 above)
Comment (puradox) [reply incubator/drone/templates/service-account.yaml]: I have changed the app label to match 👍

**Comment by timstoop** [comment]
@unguiculus can this continue or is it waiting on another fix before this one gets applied?

**Review by unguiculus — APPROVED** [review]

**Pull request merged**